Libraries/AppState/AppState.js

```diff
@@ -61,8 +61,17 @@ class AppState extends NativeEventEmitter {
     // prop and expose `getCurrentAppState` method directly.
     RCTAppState.getCurrentAppState(
       (appStateData) => {
-        if (!eventUpdated) {
+        // It's possible that the state will have changed here & listeners need to be notified
+        if (!eventUpdated && this.currentState !== appStateData.app_state) {
           this.currentState = appStateData.app_state;
+          if (this._eventHandlers['change']) {
+            this._eventHandlers['change'].forEach((subscription, key, map) => {
+              subscription.listener.apply(
```

Comment:
Instead of dispatching the event manually should we use `this.emit`?

`AppState` extends [`NativeEventEmitter`](https://github.com/facebook/react-native/blob/89757edb34cdfd22791c59981f840c7de1979325/Libraries/EventEmitter/NativeEventEmitter.js) which extends [`EventEmitter`](https://github.com/facebook/react-native/blob/8cdc7d3e0284bb6bbe22edc93a3ed23f6897837a/Libraries/vendor/emitter/EventEmitter.js) which provides an `emit` method.

Adam Comella
Microsoft Corp.